### PR TITLE
draft for doc framework

### PR DESCRIPTION
**Pull request by edublancas**
I created a short draft for the documentation framework. please share your feedback.

### CODE REVIEW

File: documentation/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+This guide explains the framework we use for writing documentation for all our projects.
+
+## Classification
+
+We can classify the resource types into four groups:
+
+- Getting started: Provide a quick route to begin using the project; their objective is to teach the basics to get started.
+- User Guide: Provide in-depth walkthroughs to use the project; they are more advanced versions than the "Getting started" guides, but their primary purpose is also teaching.
+- Cookbook: Recipes to achieve a specific goal. They provide short and to-the-point examples to perform common tasks. Their objective is to be a shortcut so the user doesn't have to gather all the pieces from reading the API reference.
```
Comment (idomic) [inline documentation/README.md:11]: I think cookbook isn't the wide term out there, maybe it should be use cases/how-to/examples/recipes.
https://docs.lakefs.io/

File: documentation/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+The following sections describe each group in more detail.
+
+### Getting started
+
+We write these tutorials as Jupyter notebooks; this makes authoring easier and allows users to download and try them with Binder. The language should be pretty straightforward, and we should omit non-essential details; the objective here is to teach the absolute basics.
+
+Depending on the project, sometimes we might decide to have a single notebook be the "Getting started" tutorial; other times, we might have multiple ones, for example, one for each module.
+
+### Guidelines
```
Comment (idomic) [inline documentation/README.md:22]: This doesn't appear in the main 4 above.

File: documentation/README.md [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+
+## Tools
+
+We're migrating all our documentation websites to [Jupyterbook](https://jupyterbook.org/en/stable/intro.html); however, some still use Sphinx.
+
+## General Guidelines
+
+### Style
+
+Writing documentation is hard, so ensure you proofread several times and use tools like Grammarly to simplify writing.
```
Comment (idomic) [inline documentation/README.md:55]: This might drop the moment we add spellcheck as part of the CI

Comment (idomic) [reply documentation/README.md:55]: I think that should be the default tbh

File: documentation/README.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+### Notebooks (applies to Getting Started and User Guide)
+
+- Notebooks should not hide any important details (e.g., hiding cells) since users are expected to copy and paste the code on them
+- In some cases (e.g., when output is too verbose), we might decide to hide some outputs, but we should be cautious of these
+- When notebooks require optional dependencies, we should document them at the top of the file so that users can install the appropriate libraries (e.g., a top cell with: `%pip install somepackage --quiet`)
+- The notebook should be organized into short sections so that users can navigate to any specific one. The titles should be short but descriptive enough
+- Since we'll run the notebooks as part of the documentation compilation process, all of them should run fast (ideally, below 10 seconds)
+- If notebooks require external files we should include both a link to the file and a command to download it (e.g., `curl -O https://ploomber.io/some-file.txt`)
```
Comment (idomic) [inline documentation/README.md:72]: Missing a point for getting started: notebooks should be short, easy to understand and without errors - this should be tested in the CI.

**Review by idomic — COMMENTED** [review]
Added comments, besides that looking good.

**Comment by idomic** [comment]
@edublancas

File: documentation/README.md (hunk 2 above)
Comment (edublancas) [reply documentation/README.md:22]: it's not one of those four, the heading level was wrong. fixed

File: documentation/README.md (hunk 3 above)
Comment (edublancas) [reply documentation/README.md:55]: spellcheck != style. we can add spellcheck, but style is also important

File: documentation/README.md (hunk 4 above)
Comment (edublancas) [reply documentation/README.md:72]: short and clear: this is mentioned in the top section
will mention the CI part

**Pull request merged**